### PR TITLE
Image view work

### PR DESCRIPTION
**Pull request by meganbkratz**
Added convenience functions to ImageView for setting colormaps and preset gradients.

### CODE REVIEW

File: pyqtgraph/graphicsItems/GradientEditorItem.py [hunk 1]
```diff
@@ -25,6 +24,14 @@
     ('grey', {'ticks': [(0.0, (0, 0, 0, 255)), (1.0, (255, 255, 255, 255))], 'mode': 'rgb'}),
 ])
 
+def addGradientListToDocstring():
+    ### create a decorator so that we can add construct a list of the gradients defined above in a docstring
+    ### Adds the list of gradients to the end of the functions docstring
```
Comment (ericdill) [inline pyqtgraph/graphicsItems/GradientEditorItem.py]: Can you rewrite this in a more standard docstring format with 
```
"""
Decorator to add a list of valid gradients to the function docstring
"""
```

Feel free to write whatever prose you want between the triple quotes, the above is merely a suggestion.

I like the very verbose function name that clearly defines what it is that this function does :smile:

File: pyqtgraph/imageview/ImageView.py [hunk 2]
```diff
@@ -717,4 +717,14 @@ def menuClicked(self):
         if self.menu is None:
             self.buildMenu()
         self.menu.popup(QtGui.QCursor.pos())
-        
+
+    def setColorMap(self, colormap):
+        """Set the color map. *colormap* is an instance of ColorMap()"""
+        self.ui.histogram.gradient.setColorMap(colormap)
+
+    def setPredefinedGradient(self, name):
```
Comment (ericdill) [inline pyqtgraph/imageview/ImageView.py]: Why not use the gradient decorator here too?

File: pyqtgraph/imageview/ImageView.py [hunk 3]
```diff
@@ -717,4 +717,14 @@ def menuClicked(self):
         if self.menu is None:
             self.buildMenu()
         self.menu.popup(QtGui.QCursor.pos())
-        
+
+    def setColorMap(self, colormap):
+        """Set the color map. *colormap* is an instance of ColorMap()"""
```
Comment (ericdill) [inline pyqtgraph/imageview/ImageView.py]: Can you write this docstring following the convention outlined [here] (https://github.com/pyqtgraph/pyqtgraph/blob/develop/CONTRIBUTING.txt#L42)?

**Comment by ericdill** [comment]
Thanks for the PR. The decorator is clever, :+1:   Just a couple of documentation related comments

**Comment by campagnola** [comment]
Merged, with a different colormap in the example.
Thanks @meganbkratz and @ericdill !

**Pull request merged**